File: Typescript/hasCycle.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function hasCycle(head: ListNode | null): boolean {
    let it1: ListNode | null = head;
    let it2: ListNode | null = head;

    while (it1 && it2 && it2.next) {
        it1 = it1.next;
        it2 = it2.next.next;
        if (it1 === it2) {
            return true;
        }
    }

    return false;
};

File: C++/hasCycle.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        auto it1 = head;
        auto it2 = head;

        while(it1 && it2 && it2->next) {
            it1 = it1->next;
            it2 = it2->next->next;
            if (it1 == it2) {
                return true;
            }
        }
        return false;
    }
};